### PR TITLE
Fixing bug when getting Hypre version. [hypre-version-dev]

### PR DESCRIPTION
**Pull request by timmoon10**
Added macros to convert HYPRE_VERSION_STRING to a string.

### CODE REVIEW

**Comment by v-dobrev** [comment]
@timmoon10,

I think the issue is that `HYPRE_RELEASE_VERSION` is defined in a different way when using `./configure`+`make` on one hand and `cmake` on the other. In the first case, the macro is quoted and the the second - it is not. So, my wild guess is that you are using CMake. :-)

With your addition, the first case will not work properly, so we need to resolve that before merging, e.g. we can check the string to see if it begins with a quote or not.

File: config/get_hypre_version.cpp [hunk 1]
```diff
@@ -18,12 +18,16 @@
 #define HYPRE_VERSION_STRING HYPRE_PACKAGE_VERSION
 #endif
 
+// Macros to expand a macro as a string
+#define STR_EXPAND(s) #s
+#define STR(s) STR_EXPAND(s)
```
Comment (v-dobrev) [inline config/get_hypre_version.cpp]: Why not just define:
```c++
#define STR(s) #s
```

Comment (timmoon10) [inline config/get_hypre_version.cpp]: According to Google, we need to use the two-step process to properly expand the macro.

Comment (v-dobrev) [inline config/get_hypre_version.cpp]: OK, Google is right (at least this time). :-) I guess the explanation is that we are using `STR()` with another macro, so it has to be expanded twice.

Comment (tzanio) [inline config/get_hypre_version.cpp]: That's just weird...

**Pull request merged**